manual/manual_circuit.rst

```diff
@@ -435,36 +435,168 @@ In addition :py:meth:`Circuit.get_unitary` can be used to numerically calculate
+
+As characteristics of a :py:class:`Circuit` go, these are pretty basic. In terms of approximating the noise level, they fail heavily from weighting all gates evenly when, in fact, some will be much harder to implement than others. For example, in the NISQ era, we find that most technologies provide good single-qubit gate times and fidelities, with two-qubit gates being much slower and noisier [Arut2019]_. On the other hand, looking forward to the fault-tolerant regime we will expect Clifford gates to be very cheap but the magic :math:`T` gates to require expensive distillation procedures [Brav2005]_ [Brav2012]_.
+
+We can use the :py:class:`OpType` enum class to look for the number of gates of a particular type. Additionally, the methods :py:meth:`n_1qb_gates`, :py:meth:`n_2qb_gates` and :py:meth:`n_nqb_gates` can be used to count the number of gates in terms of how many qubits they act upon irrespective of type.
+
+We also define :math:`G`-depth (for a subset of gate types :math:`G`) as the minimum number of layers of gates in :math:`G` required to run the :py:class:`Circuit`, allowing for topological reorderings. Specific cases of this like :math:`T`-depth and :math:`CX`-depth are common to the literature on circuit simplification [Amy2014]_ [Meij2020]_.
+
+.. jupyter-execute::
+
+    from pytket import Circuit, OpType
+    from pytket.circuit.display import render_circuit_jupyter
+
+    circ = Circuit(3)
+    circ.T(0)
+    circ.CX(0, 1)
+    circ.CX(2, 0)
+    circ.add_gate(OpType.CnRy, [0.6], [0, 1, 2])
+    circ.T(2)
+    circ.CZ(0, 1)
+    circ.CZ(1, 2)
+    circ.T(1)
+
+    render_circuit_jupyter(circ) # draw circuit diagram
+
+    print("T gate count =", circ.n_gates_of_type(OpType.T))
+    print("#1qb gates =", circ.n_1qb_gates())
+    print("#2qb gates =", circ.n_2qb_gates())
+    print("#3qb gates =", circ.n_nqb_gates(3)) # count the single CnRy gate (n=3)
+    print("T gate depth =", circ.depth_by_type(OpType.T))
+    print("2qb gate depth =", circ.depth_by_type({OpType.CX, OpType.CZ}))
+
+.. note:: Each of these metrics will analyse the :py:class:`Circuit` "as is", so they will consider each Box as a single unit rather than breaking it down into basic gates, nor will they perform any non-trivial gate commutations (those that don't just follow by deformation of the DAG) or gate decompositions (e.g. recognising that a :math:`CZ` gate would contribute 1 to :math:`CX`-count in practice).
+
 Boxes
 -----
 
-.. Boxes abstract away complex structures as black-box units within larger circuits
-
 Working with individual basic gates is sufficient for implementing arbitrary circuits, but that doesn't mean it is the most convenient option. It is generally far easier to argue the correctness of a circuit's design when it is constructed using higher-level constructions. In ``pytket``, the concept of a "Box" is to abstract away such complex structures as black-boxes within larger circuits.
 
+Circuit Boxes and Controlled Unitaries
```

Comment:
Seems a bit odd to group these together, why not make separate sections? Also "controlled unitaries" isn't the best description for a `QControlBox` since the constructor takes an `Op`, not a unitary.

Reply:
Out of curiosity is it even possible to construct a controlled non-unitary operation in pytket? The manual says the following...

```
If our subcircuit is a pure quantum circuit (i.e. it corresponds to a unitary operation), we can construct the controlled version that is applied coherently according to some set of control qubits.
```

Reply:
No it's not possible. (I'm not even sure what it would mean, but there may be a way to make sense of it e.g. using density matrices?)